docs/reference/contributing/guidelines/workflow.md

```diff
@@ -93,13 +93,13 @@ We consider the following pull request types.
 
 #### Fix
 
-A bug fix is a backward-compatible internal change that fixes incorrect behavior.
+A bug fix is a change that fixes a specific defect in the codebase with backwards compatibility. These are the highest priority because of the positive impact the change will have on users developing against the same code. A bug fix should be limited to restoring the documented or proven otherwise, originally intended behavior. Bug fixes are candidates for patch releases.
 
 Release: patch
 
 #### Refactor
 
-Refactors are intended for feature releases, if they are not fixing specific issues, because they can introduce new issues.
+A refactor is a contribution that modifies the codebase without fixing a bug or changing the existing behavior. Examples of this would be moving functions or variables between translation units, renaming source files or folders, scope modification for nonpublic code, documentation structure changes, and test organization changes. There is always the risk that someone depended on the location or name before a refactor therefore these are lower in priority than bug fixes and might require detailed justification for the change. Refactors are candidates for feature releases. 
```

Comment:
If a refactor is made in a backwards-compatble way, can they be targeted for patch releases?

Reply:
What about if a refactor fixes a bug?

Reply:
Either they fix a bug separately or will go with refactor. Everything should be backward compatible (only exceptionally not)

Reply:
So making sure, if a the only way to fix a particular bug is with a refactor, that PR will have to come into the next major release?

Reply:
It should